store/store.go

```diff
@@ -325,6 +309,43 @@ func (s *Store) Remove(key []byte) (bool, error) {
 	return removed, nil
 }
 
+func (s *Store) getPrimaryKeyData(blk types.Block, indexKey []byte) ([]byte, []byte, error) {
+	// Get the key and value stored in primary to see if it is the same (index
+	// only stores prefixes).
+	storedKey, storedValue, err := s.index.Primary.Get(blk)
+	if err != nil {
+		// Log the error reading the primary, since no error is returned if the
+		// bad index is sucessfully deleted.
+		log.Errorw("Error reading primary, removing bad index", "err", err)
```

Comment:
Worth logging more information here, like the `indexKey` or something about the block?

Reply:
The indexKey is a digest of a multihash, so probably not really meaningful.  The really is not much available to log here.  The most important thing is to be able to determine if this error is happening, meaning it is possible some data may be missing.